protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -676,8 +676,18 @@ class _FieldSet {
       fieldValue = _cloneMessage(fieldValue);
     }
     if (fi.index == null) {
+      final current = _ensureExtensions()._getFieldOrNull(fi);
+      if (current != null) {
+        fieldValue = current..mergeFromMessage(fieldValue);
```

Comment:
In this case I don't think the clone is needed. (and actually harmful, leads to O(n^2) behavior